extmod/modopenamp_remoteproc.c

```diff
@@ -0,0 +1,175 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2023-2024 Arduino SA
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * OpenAMP's remoteproc class.
+ */
+
+#if MICROPY_PY_OPENAMP_REMOTEPROC
+
+#include "py/obj.h"
+#include "py/nlr.h"
+#include "py/runtime.h"
+#include "py/stream.h"
+#include "extmod/vfs.h"
+
+#include "metal/sys.h"
+#include "metal/alloc.h"
+#include "metal/errno.h"
+#include "metal/io.h"
+
+#include "openamp/open_amp.h"
+#include "openamp/remoteproc.h"
+#include "openamp/remoteproc_loader.h"
+
+#include "modopenamp.h"
+#include "modopenamp_remoteproc.h"
+
+#define DEBUG_printf(...)   // mp_printf(&mp_plat_print, __VA_ARGS__)
+
+#if !MICROPY_PY_OPENAMP
+#error "MICROPY_PY_OPENAMP_REMOTEPROC requires MICROPY_PY_OPENAMP"
+#endif
+
+typedef struct openamp_remoteproc_obj {
+    mp_obj_base_t base;
+    struct remoteproc rproc;
+} openamp_remoteproc_obj_t;
+
+const mp_obj_type_t openamp_remoteproc_type;
+
+// Port-defined image store operations.
+extern struct image_store_ops openamp_remoteproc_store_ops;
+
+// Port-defined remote-proc operations.
+const struct remoteproc_ops openamp_remoteproc_ops = {
+    .init = mp_openamp_remoteproc_init,
```

Comment:
@dpgeorge do we have to keep the `mp_` prefix here ? and in the port's implementation.

Reply:
I think it's fine having `mp_` prefix, because it's a global symbol.  It's consistent with, eg, all the functions in `ports/stm32/mpnimbleport.c`.

Reply:
Okay no problem.